src/model/priors/single_cell_bulk.rs

```diff
@@ -142,14 +142,12 @@ impl PairModel<DiscreteAlleleFreqs, ContinuousAlleleFreqs> for SingleCellBulkMod
         };
         let k_single = 0..n_obs_s + 1;
 
-        let k_start = if *af_bulk.start == 0.0 { // 0 as a bulk range start is always inclusive
-            0 as u64
-        } else {
-            // any non-zero bulk range start is always exclusive
-            (*af_bulk.start * n_obs_bulk as f64).floor() as u64 + 1
-        };
-        // any bulk range end is always inclusive
-        let k_end = (*af_bulk.end * n_obs_bulk as f64).floor() as u64 + 1;
+        // stable rust ranges currently only allow inclusive start points, for exclusive add
+        // 0.00000001 in the Event definition
+        let k_start = (*af_bulk.start * n_obs_bulk as f64).ceil() as u64;
+        // stable rust ranges currently only allow exclusive end points, for including an end point,
+        // add 0.00000001 in the Event definition
+        let k_end = (*af_bulk.end * n_obs_bulk as f64).floor() as u64;
```

Comment:
Sure, that looks like a viable solution. I also considered implementing something, but then figured it would be worth waiting for std::ops::RangeInclusive to become stable for a proper solution and instead going for the easiest hack in the meantime. But then I realise, that the new std::ops::Range also doesn't account for an exclusive start, yet. So we would have to still work around this in the future -- maybe with something like `std::ops::Range.skip_while(|x| x == start)` at range definition, although I'm not sure about types (range vs. resulting iterator). So maybe we need to implement this ourselves, after all.

Currently I'm doing some analysis with this state of the single-cell-bulk branch and current ProSolo, so I'd leave this open for now and address it in the coming weeks.